src/services/services.ts

```diff
@@ -2937,38 +2896,75 @@ module ts {
-                //               We COULD also just do what 'getSymbolModifiers' does, which is to use the first declaration.
-                Debug.assert(session.typeChecker.getTypeOfSymbolAtLocation(symbol, location) !== undefined, "Could not find type for symbol");
-                let displayPartsDocumentationsAndSymbolKind = getSymbolDisplayPartsDocumentationAndSymbolKind(symbol, getValidSourceFile(fileName), location, session.typeChecker, location, SemanticMeaning.All);
-                return {
-                    name: entryName,
-                    kind: displayPartsDocumentationsAndSymbolKind.symbolKind,
-                    kindModifiers: completionEntry.kindModifiers,
-                    displayParts: displayPartsDocumentationsAndSymbolKind.displayParts,
-                    documentation: displayPartsDocumentationsAndSymbolKind.documentation
-                };
+            var entries = getCompletionEntriesFromSymbols(symbols);
+
+            // Add keywords if this is not a member completion list
+            if (!isMemberCompletion) {
+                addRange(entries, keywordCompletions);
             }
-            else {
-                // No symbol, it is a keyword
+
+            return { isMemberCompletion, isNewIdentifierLocation, entries };
+
+            function getCompletionEntriesFromSymbols(symbols: Symbol[]): CompletionEntry[] {
+                let start = new Date().getTime();
+                var entries: CompletionEntry[] = [];
+                var nameToSymbol: Map<Symbol> = {};
+
+                for (let symbol of symbols) {
+                    let entry = createCompletionEntry(symbol, typeInfoResolver, location);
+                    if (entry) {
+                        let id = escapeIdentifier(entry.name);
+                        if (!lookUp(nameToSymbol, id)) {
+                            entries.push(entry);
+                            nameToSymbol[id] = symbol;
+                        }
+                    }
+                }
+                log("getCompletionsAtPosition: getCompletionEntriesFromSymbols: " + (new Date().getTime() - start));
+                return entries;
+            }
+        }
+
+        function getCompletionEntryDetails(fileName: string, position: number, entryName: string): CompletionEntryDetails {
+            synchronizeHostData();
+
+            // Compute all the completion symbols again.
+            let completionData = getCompletionData(fileName, position);
+            if (completionData) {
+                let { symbols, location } = completionData;
+
+                // Find the symbol with the matching entry name.
+                let target = program.getCompilerOptions().target;
+                let symbol = forEach(symbols, s => getCompletionEntryDisplayName(s, target, /*performCharacterChecks:*/ false) === entryName ? s : undefined);

```

Comment:
Explain that you don't need to perform the validation check because `entryName` is anticipated to be a valid name.